src/components/sbb-checkbox-group/sbb-checkbox-group.tsx

```diff
@@ -1,136 +1,120 @@
       toggleDatasetEntry(checkbox, 'groupDisabled', this.disabled);
     }
   }
 
-  @Watch('required')
-  public updateRequired(): void {
+  private _updateRequired(): void {
     for (const checkbox of this._checkboxes) {
       toggleDatasetEntry(checkbox, 'groupRequired', this.required);
     }
   }
 
-  @Watch('size')
-  public updateSize(): void {
+  private _updateSize(): void {
     for (const checkbox of this._checkboxes) {
       checkbox.size = this.size;
     }
   }
 
-  public connectedCallback(): void {
-    toggleDatasetEntry(
-      this._element,
-      'hasSelectionPanel',
-      !!this._element.querySelector('sbb-selection-panel'),
-    );
+  public override connectedCallback(): void {
+    super.connectedCallback();
+    const signal = this._abort.signal;
+    this.addEventListener('keydown', (e) => this._handleKeyDown(e), { signal });
+    toggleDatasetEntry(this, 'hasSelectionPanel', !!this.querySelector('sbb-selection-panel'));
     this._handlerRepository.connect();
   }
 
-  public disconnectedCallback(): void {
+  public override willUpdate(changedProperties: PropertyValues<this>): void {
+    if (changedProperties.has('disabled')) {
+      this._updateDisabled();
+    }
+    if (changedProperties.has('required')) {
+      this._updateRequired();
+    }
+    if (changedProperties.has('size')) {
+      this._updateSize();
+    }
+  }
+
+  public override disconnectedCallback(): void {
+    super.disconnectedCallback();
     this._handlerRepository.disconnect();
   }
 
-  @Listen('keydown')
-  public handleKeyDown(evt: KeyboardEvent): void {
-    const enabledCheckboxes: HTMLSbbCheckboxElement[] = this._checkboxes.filter(
-      (checkbox: HTMLSbbCheckboxElement) =>
+  private _handleKeyDown(evt: KeyboardEvent): void {
+    const enabledCheckboxes: SbbCheckbox[] = this._checkboxes.filter(
+      (checkbox: SbbCheckbox) =>
         !isValidAttribute(checkbox, 'disabled') && interactivityChecker.isVisible(checkbox),
     );
 
     if (
       !enabledCheckboxes ||
       // don't trap nested handling
-      ((evt.target as HTMLElement) !== this._element &&
-        (evt.target as HTMLElement).parentElement !== this._element &&
+      ((evt.target as HTMLElement) !== this &&
+        (evt.target as HTMLElement).parentElement !== this &&
         (evt.target as HTMLElement).parentElement.nodeName !== 'SBB-SELECTION-PANEL')
     ) {
       return;
     }
 
     if (isArrowKeyPressed(evt)) {
-      const current: number = enabledCheckboxes.findIndex(
-        (e: HTMLSbbCheckboxElement) => e === evt.target,
-      );
+      const current: number = enabledCheckboxes.findIndex((e: SbbCheckbox) => e === evt.target);
       const nextIndex: number = getNextElementIndex(evt, current, enabledCheckboxes.length);
       enabledCheckboxes[nextIndex]?.focus();
     }
   }
 
   private _updateCheckboxes(): void {
-    const checkboxes = this._checkboxes;
+    const checkboxes: SbbCheckbox[] = this._checkboxes;
```

Comment:
nitpick, but this should not be necessary
```suggestion
    const checkboxes = this._checkboxes;
```